examples/Titanic/titanic_handler.py

```diff
@@ -0,0 +1,158 @@
+        if os.path.isfile(mapping_file_path):
+            print("Mapping file present")
+            with open(mapping_file_path) as f:
+                self.mapping = json.load(f)
+        else:
+            print("Mapping file missing")
+            logger.warning("Missing the index_to_name.json file.")
+
+        # ------------------------------- Captum initialization ----------------------------#
+        self.ig = IntegratedGradients(self.model)
+        self.initialized = True
+
+    def preprocess(self, data):
+        """Basic text preprocessing, based on the user's chocie of application mode.
+
+        Args:
+            data (csv): The Input data in the form of csv is passed on to the preprocess
+            function.
+
+        Returns:
+            list : The preprocess function returns a list of Tensor and feature names
+        """
+        self.input_file_path = data[0]["body"]["input_file_path"][0]
+        data = pd.read_csv(self.input_file_path)
+        self.feature_names = list(data.columns)
+        data = data.to_numpy()
+        data = torch.from_numpy(data).type(torch.FloatTensor)
+        return data
+
+    def inference(self, data):
+        """Predict the class (survived or not survived) of the received input json file using the
+        serialized model.
+
+        Args:
+            input_batch (list): List of Tensors from the pre-process function is passed here
+
+        Returns:
+            list : It returns a list of the predicted value for the input test record
+        """
+        self.out_probs = self.model(data).detach().numpy()
+        self.predicted_idx = self.out_probs.argmax(1).item()
+        prediction = self.mapping[str(self.predicted_idx)]
+        self.inference_output.append(prediction)
+        logger.info("Model predicted: '%s'", prediction)
+        return [prediction]
+
+    def postprocess(self, inference_output):
+        """Post Process Function converts the predicted response into Torchserve readable format.
+
+        Args:
+            inference_output (list): It contains the predicted response of the input record.
+        Returns:
+            (list): Returns a list of the Predictions and Explanations.
+        """
+        return inference_output
+
+    def get_insights(
+        self, input_tensor, title="Average Feature Importances", axis_title="Features"
+    ):
+        """This function calls the integrated gradient to the feature importance
+
+        Args:
+            data(tensor):
+            target (int): The Target can be set to any acceptable label under the user's discretion.
+
+        Returns:
+            (list): Returns a dict of feature names and their importances
+        """
+        ig = IntegratedGradients(self.model)
+        input_tensor.requires_grad_()
+        attr, self.delta = ig.attribute(input_tensor, target=1, return_convergence_delta=True)
+        attr = attr.detach().numpy()
```

Comment:
@Arvind-Ideas2IT as input has been moved to device, need to move it to cpu for converting to numpy, attr = attr.cpu().detach().numpy()

Reply:
@HamidShojanazeri Got it,Thanks. I will update these changes.